setup.cfg

```diff
@@ -100,10 +100,20 @@ doctest_plus = enabled
 text_file_format = rst
 open_files_ignore = "astropy.log" "/etc/hosts" "*.ttf"
 remote_data_strict = true
-addopts = -p no:warnings --doctest-rst
+addopts = --doctest-rst
 asdf_schema_root = astropy/io/misc/asdf/data/schemas
 xfail_strict = true
 qt_no_exception_capture = 1
+filterwarnings =
+    error
+    ignore:unclosed file:ResourceWarning
+    ignore:unclosed <socket:ResourceWarning
+    ignore:unclosed <ssl.SSLSocket:ResourceWarning
+    ignore:The timefunc function is deprecated
+    ignore:numpy.ufunc size changed:RuntimeWarning
+    ignore:Remote data cache could not be accessed due to NotADirectoryError
+    ignore:.*Cache directory cannot be read or created
+    ignore:can't resolve package from __spec__
```

Comment:
Are you sure we need this, @astrofrog ? Do you see it when you run with `python setup.py test` instead of `pytest` directly? I thought we have suppressed it either in

https://github.com/astropy/astropy/blob/93901211dc3c5074325068604b5b69e0c93e1db2/astropy/tests/helper.py#L131

or

https://github.com/astropy/astropy/blob/93901211dc3c5074325068604b5b69e0c93e1db2/astropy/tests/helper.py#L149-L151

Reply:
I saw it when running the tests for the docs with ``pytest docs`` which bypasses the stuff in astropy/tests/helper.py. Ideally we should just define all the things to ignore in ``setup.cfg`` now rather than in ``astropy/tests/helper.py``. So can we leave this entry in ``setup.cfg``, and I'll look shortly (after this PR) into whether we can combine some of the stuff in ``helper.py`` with ``setup.cfg``?

Reply:
@astrofrog , if you want to take a stab at it, there is also some incantations in `astropy/conftest.py`.

https://github.com/astropy/astropy/blob/93901211dc3c5074325068604b5b69e0c93e1db2/astropy/conftest.py#L25

Reply:
yes, this should not be in `astropy/astropy/tests/helper.py` as that file will be picked up by other packages, too that use the test runner, and they may or may not want to ignore all of these (in practice they probably want to but you get the point)

Reply:
If you remove it now, they are gonna suddenly see all these warnings... 🤷‍♀

Reply:
my point is that they should rather live in the conftest, shouldn't they?

Reply:
I think Tom R wants to move them out to `setup.cfg` altogether.

Reply:
that's out of the scope I believe, wouldn't that be a new feature after feature freeze? :trollface:

Reply:
(btw, what's wrong with conftest.py as opposed to setup.cfg?)

Reply:
Definitely out of scope for this PR! Let's chat about this at the coordination meeting, it's not urgent. I was just trying to justify adding a warning to ignore to ``setup.cfg``. The short answer @bsipocz is that if you want to be able to do things like ``pytest docs/coordinates`` then ``astropy/conftest.py`` is ignored.